examples/goldilocks_models/find_goldilocks_models.cc

```diff
@@ -1664,20 +1717,41 @@ int findGoldilocksModels(int argc, char* argv[]) {
   }
   const int method_to_solve_system_of_equations =
       FLAGS_method_to_solve_system_of_equations;
-  // With bigger momentum, you might need a larger tolerance
+  // With bigger momentum, you might need a larger tolerance.
+  // If not uniform grid, considering sample cost increase makes no sense.
+  // So, use a very large tolerance on increase rate.
   double max_sample_cost_increase_rate = 0;
   if (FLAGS_robot_option == 0) {
-    max_sample_cost_increase_rate = FLAGS_is_stochastic? 2.0: 0.01;
+      if(uniform_grid){
+          max_sample_cost_increase_rate = FLAGS_is_stochastic? 2.0: 0.01;
+      }else{
+          max_sample_cost_increase_rate = 100;
+      }
   } else if (FLAGS_robot_option== 1) {
-    max_sample_cost_increase_rate = FLAGS_is_stochastic? 0.5: 0.01; //0.3
+      if(uniform_grid){
+          max_sample_cost_increase_rate = FLAGS_is_stochastic? 0.5: 0.01; //0.3
+      }else{
+          max_sample_cost_increase_rate = 100;
+      }
   } else {
     throw std::runtime_error("Should not reach here");
   }
+  /// Increase the tolerance for restricted number
+  /// Maybe we can turn off the shrinking step size when using restricted number of sample
+  /// Didn't see improvement after shrinking the step size.
   double max_average_cost_increase_rate = 0;
   if (FLAGS_robot_option == 0) {
     max_average_cost_increase_rate = FLAGS_is_stochastic? 0.5: 0.01;
+    if(restricted_sample_number)
+    {
```

Comment:
Although I haven't apply Google code style to all of my code, I mostly try to follow it. 
I notice some of your code style is different from Google's. If you are using CLion, you can run Drake's external tools to auto-format the code: https://drake.mit.edu/clion.html#integrating-external-tools-with-clion
If you are not using the tools and think it isn't worth your time, I can reformat the code (and push to `set-initial-guess` branch) right before I merge your branch into mine

Reply:
I just set the code style in the setting of CLion to google format style. I think it is necessary to make the code format follow a common style. So please let me know if there is still any irregular style.